internal/model.go

```diff
@@ -0,0 +1,229 @@
+/**
+ *  Copyright (c) 2020  Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package internal
+
+import (
+	"context"
+)
+
+const (
+
+	// plugin types
+	//TODO: where should this live?
+
+	PluginTypeKafka  = "kafka"  // generic kafka plugin
+	PluginTypeKDS    = "kds"    // generic kds plugin
+	PluginTypeSQS    = "sqs"    // generic sqs plugin
+	PluginTypeHTTP   = "http"   // generic http / webhook plugin
+	PluginTypeEars   = "ears"   // loopback plugin, may be useful for multi stage processing
+	PluginTypeGears  = "gears"  // deliver to kafka in masheens envelope
+	PluginTypeNull   = "null"   // black whole for events for testing
+	PluginTypeDebug  = "debug"  // debug plugin
+	PluginTypeFilter = "filter" // a filter plugin that filters or transforms and event
```

Comment:
I think Trevor should chime in, but this feels more like a plugin thing and we shouldn't define it here in ears core